torch/_inductor/fx_passes/post_grad.py

```diff
@@ -1070,19 +1079,29 @@ def move_constructors_to_cuda(graph):
 
         constructors.append(node)
 
-    # not handling multiple cuda devices initially
-    if not constructors or len(cuda_devices) != 1:
-        return
+    return constructors, cuda_devices
 
+
+def move_constructors_to_cuda_impl(graph_module):
+    graph = graph_module.graph
+    constructors, cuda_devices = gather_constructors_and_cuda_devices(graph)
     movable_constructors = find_movable_constructors(graph, constructors)
 
+    if len(movable_constructors) == 0:
+        return
+
+    if len(cuda_devices) != 1:
+        raise ZeroOrMultipleDevicesError("cuda", cuda_devices)
```

Comment:
This pass shouldn't error, it's just a missed optimization if there are multiple devices.  Can we remove this ? 

In general, this pass has no guarantee of moving all compute away from cpu. If there are cpu inputs or outputs, or non movable operators, this won't succed.

The xla pass can do a follow up pass to see if it was successful.